plugins/fastly/api_token.go

```diff
@@ -0,0 +1,77 @@
+package fastly
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIToken,
+		DocsURL:       sdk.URL("https://docs.fastly.com/en/guides/using-api-tokens"),
+		ManagementURL: sdk.URL("https://manage.fastly.com/account/personal/tokens"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "API Token used to authenticate to Fastly.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 32,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryAllEnvVars(fieldname.Token, "FASTLY_API_TOKEN"),
+			importer.MacOnly(
+				TryFastlyConfigFile("~/Library/Application Support/fastly/config.toml"),
```

Comment:
if fastly's config file location depends on `XDG_CONFIG_HOME` I would first check here if that env var is set to anything, and if yes, use that location, otherwise use the already existing logic.

Reply:
I would suggest not to do that. We're planning to add `XDG_CONFIG_FILE` support for all plugins at once: https://github.com/1Password/shell-plugins/issues/153

Easiest would be to use the hardcoded location for now, as it is currently in place. This would ensure that different plugins don't use diverging standards for handling this variable, which would make the migration harder, later on. 😄

Let me know if you disagree, @AndyTitu

Reply:
Resolving